synapse/rest/client/v2_alpha/register.py

```diff
@@ -677,18 +689,25 @@ async def on_POST(self, request):
 
         return 200, return_dict
 
-    async def _do_appservice_registration(self, username, as_token, body):
+    async def _do_appservice_registration(
+        self, username, as_token, body, should_issue_refresh_token: bool = False
+    ):
         user_id = await self.registration_handler.appservice_register(
             username, as_token
         )
         return await self._create_registration_details(
             user_id,
             body,
             is_appservice_ghost=True,
+            should_issue_refresh_token=should_issue_refresh_token,
         )
 
     async def _create_registration_details(
-        self, user_id, params, is_appservice_ghost=False
+        self,
+        user_id,
+        params,
+        is_appservice_ghost=False,
+        should_issue_refresh_token: bool = False,
```

Comment:
docstring for the new param please. (while you're here, perhaps you could move the types from the docstring into annotations on the params.)

(NB `params` is a `dict`, not an `object` - looks like someone had been writing too much javascript...)

Reply:
Done in https://github.com/matrix-org/synapse/pull/9450/commits/6024ed89983c0cf756a354279cf9eb464249acc1